### PR TITLE
[SWP-1609] [ALT] Extract VAT logic into i18n package

### PR DESCRIPTION
**Pull request by clemblanco**

### CODE REVIEW

File: src/Facades/CurrencyHelper.php [hunk 1]
```diff
@@ -2,13 +2,11 @@
...
 namespace PodPoint\I18n\Facades;
 
-use Illuminate\Support\Facades\Facade;
-
 /**
  * @method static string getSymbol()
  * @method static string toFormat()
  */
-class CurrencyHelper extends Facade
+class CurrencyHelper extends \Illuminate\Support\Facades\Facade
```
Comment (lukasz-kowal) [inline src/Facades/CurrencyHelper.php:9]: Why did you change this?

File: src/TaxRate.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Returns the tax rate for the given country code.
+     *
+     * @param string $countryCode
+     *
+     * @return float
+     */
+    public function get(string $countryCode): float
+    {
+        return (float) $this->vatCalculator->getTaxRateForCountry($countryCode);
```
Comment (lukasz-kowal) [inline src/TaxRate.php:28]: is there need for `(float)` ?
getTaxRateForCountry apparently returns float and even if, php return type annotation should get the job done anyway.

Comment (lukasz-kowal) [reply src/TaxRate.php:28]: this will apply for the other methods as well

File: src/TaxRate.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     *
+     * @param int|float $netPrice The net price to use for the calculation
+     * @param string $countryCode The country code to use for the rate lookup
+     * @param string|null $postalCode The postal code to use for the rate exception lookup
+     * @param bool|null $company Whether or not the customer is a company
+     * @param string|null $type The type can be low or high
+     *
+     * @return float
+     */
+    public function calculate($netPrice, string $countryCode, $postalCode = null, $company = null, $type = null): float
```
Comment (lukasz-kowal) [inline src/TaxRate.php:43]: I would add types to all arguments

File: src/TaxRate.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     *
+     * @param int|float $grossPrice The gross price to use for the calculation
+     * @param string $countryCode The country code to use for the rate lookup
+     * @param string|null $postalCode The postal code to use for the rate exception lookup
+     * @param bool|null $company Whether or not the customer is a company
+     * @param string|null $type The type can be low or high
+     *
+     * @return float
+     */
+    public function exclude($grossPrice, string $countryCode, $postalCode = null, $company = null, $type = null): float
```
Comment (lukasz-kowal) [inline src/TaxRate.php:60]: missing typings

File: src/Facades/TaxRate.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace PodPoint\I18n\Facades;
+
+/**
+ * @method static float get(string $countryCode)
+ * @method static float calculate(int|float $netPrice, string $countryCode, string|null $postalCode, bool|null $company, string|null $type)
+ * @method static float exclude(int|float $netPrice, string $countryCode, string|null $postalCode, bool|null $company, string|null $type)
+ */
+class TaxRate extends \Illuminate\Support\Facades\Facade
```
Comment (lukasz-kowal) [inline src/Facades/TaxRate.php:10]: why using full path?

Comment (clemblanco) [reply src/Facades/TaxRate.php:10]: Laravel Facades use full paths. Saves one line ¯\_(ツ)_/¯

File: src/TaxRate.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+     *
+     * @param int|float $netPrice The net price to use for the calculation
+     * @param string $countryCode The country code to use for the rate lookup
+     * @param string|null $postalCode The postal code to use for the rate exception lookup
+     * @param bool|null $company Whether or not the customer is a company
+     * @param string|null $type The type can be low or high
+     *
+     * @return float
+     */
+    public function calculate($netPrice, string $countryCode, string $postalCode = null, bool $company = null, string $type = null): float
```
Comment (lukasz-kowal) [inline src/TaxRate.php:43]: int|float 🤦

File: src/TaxRate.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    /**
+     * Returns the tax rate for the given country code.
+     *
+     * @param string $countryCode
+     *
+     * @return float
+     */
+    public function get(string $countryCode): float
```
Comment (lukasz-kowal) [inline src/TaxRate.php:26]: Out of curiosity why you are adding return type hinting here but not in other places?

File: src/TaxRate.php (hunk 6 above)
Comment (clemblanco) [reply src/TaxRate.php:43]: https://github.com/mpociot/vat-calculator/blob/master/src/Mpociot/VatCalculator/VatCalculator.php#L457

¯\_(ツ)_/¯

Comment (lukasz-kowal) [reply src/TaxRate.php:43]: yeah I have seen it but still...
Hence I don't like doc block typings, completely useless.

File: src/TaxRate.php (hunk 7 above)
Comment (clemblanco) [reply src/TaxRate.php:26]: You're right. Inconsistency. Removing them for now until we settle on some standards as a team enforced with linting and CI pipelines.

Comment (lukasz-kowal) [reply src/TaxRate.php]: Please keep then!
I was just asking why?
They are far more important than doc blocks and they will actually cast return type.

Comment (clemblanco) [reply src/TaxRate.php:26]: Added them everywhere. I still think docblocks can let us do more like `array|null`. We'll need to wait until we upgrade all our stacks to php 7.3+ to get a better support of typehinting IMO.

Comment (lukasz-kowal) [reply src/TaxRate.php:26]: you can do array|null even now?
`?array $someArray = null`
I think atm you cannot do things you should not be doing?
